hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieCompactionConfig.java

```diff
@@ -266,6 +272,20 @@ public class HoodieCompactionConfig extends HoodieConfig {
       .sinceVersion("0.11.0")
       .withDocumentation("When rewriting data, preserves existing hoodie_commit_time");
 
+  public static final ConfigProperty<String> PARTITIONS_FOR_COMPACTION = ConfigProperty
```

Comment:
Log compaction we are looking to run on record index partitions for metadata table. The other partitions of the metadata table like file listing partition may not need it. So, we thought we can target specific partitions.
I understand this might cause some issues in archival when logcompaction is working only on few partitions. I can remove these configs.

Reply:
I see. So the main use case here is compaction of record index changes. Seems a little specific to have a top level config for it. i would rather not have this introduced.

Reply:
Instead of explicitly specifying that log compaction is allowed only on say record_index partition and not on files partition, should we have the ability to run log-compaction on any of the partition, but whether log-compaction gets triggered could be based on some other criteria.  (compact only if log file blocks contain K records, or has size > X etc).